### DISCUSSION TITLE
Generate request validation exception response based on the Request object

### DISCUSSION THREAD

**Discussion by szana8** [dedoc/scramble#46] [Ideas]
Hi,

If it is possible it would be great to have another response when a Laravel Request object has been used in the controller for request validation. 
For example a CreateUserRequest contains some rules and messages to check the request. If it fails Laravel automatically return with an exception based on the rule messages. This is really important to put this information into the API doc.

Thanks,

**Comment by romalytvynenko** [comment]
Hey @szana8! That definitely makes sense.

Can you please tell me more about your expectations?

Do you simply want to know that there may be 422 response, or you would like to see its structure as well? 🧐

**Reply by romalytvynenko** [reply]
Would be cool if you already had some docs in place and can share an example 😇

**Reply by szana8** [reply]
Yea the response code have to be 422 and some information in the description about the error.  Like in here: https://swagger.io/docs/specification/describing-responses/
And also would be great to be able to manually add some other exceptions and description of those via annotation. Like:
/**
 * @exception('The description of the exception', code_of_the_exception)
*/ 
Because many times the routes can return back with Authenticatio, Authorization and other exceptions too.

**Reply by romalytvynenko** [reply]
Cool, thanks. I guess Scramble can handle some of that (422 and 403 for sure).

Will write here once done.

**Comment by romalytvynenko** [comment]
Closing as was added to Scramble
